src/Components/src/Microsoft.AspNetCore.Components/Rendering/Renderer.cs

```diff
@@ -22,6 +23,14 @@ public abstract class Renderer : IDisposable
         private int _nextComponentId = 0; // TODO: change to 'long' when Mono .NET->JS interop supports it
         private bool _isBatchInProgress;
         private int _lastEventHandlerId = 0;
+        private List<Task> _pendingTasks = new List<Task>();
```

Comment:
I might be missing something, but it looks like in the non-HtmlRenderer case, this list will grow infinitely and is never cleared after the initial render.

Should we do something so that only HtmlRenderer actually populates this list, and the others do nothing in `AddToPendingTasks`? The `AddToPendingTasks` could be changed to a virtual method that no-ops in the default case.

Reply:
This gets cleaned up in all code paths as 
```csharp
void RenderRootComponent(int componentId, ParameterCollection initialParameters)
{
            ReportAsyncExceptions(RenderRootComponentAsync(componentId, initialParameters));
}
```

Internally makes sure that all the cleanup happens.

Reply:
I thought that only happened on the first render cycle. When some event like a button click happens later in the other renderers, this code path doesn't execute, does it?

Reply:
Ah, I see what you mean. You're right. But it has an easy fix

Reply:
I null the list after RenderRootComponentAsync is done